SwiftWisdom/Core/StandardLibrary/Array/Array+Utilities.swift

```diff
@@ -32,6 +32,37 @@ extension Array {
     }
 }
 
+extension Array {
+    /// Separates this array's elements into chunks with a given length.
+    ///
+    /// If `length` is not a positive number or if it's greater than or equal to this array's length, we return a new array
+    /// whose sole element is this array itself.
+    ///
+    /// - parameter length: Maximum number of elements in each chunk.
+    /// - parameter includeRemainder: If set to `true`, include the last chunk in returned array even if its length is less than
+    ///   `length`. If set to `false`, discard the last chunk if its length is less than `length`. Defaults to `true`.
+    ///
+    /// - returns: New array of chunks of elements in this array.
+    public func ip_chunks(ofLength length: Int, includeRemainder: Bool = true) -> [[Element]] {
+        guard length > 0 else { return [self] }
+
+        var mutable = self
+        var chunks = [[Element]]()
+
+        while mutable.count >= length {
+            let next = Array(mutable.prefix(length))
+            chunks.append(next)
+            mutable.removeFirst(length)
```

Comment:
You're right! In hindsight making `remove*` calls seems like a bad idea especially for very large arrays. I propose that we document that the order of the elements is preserved (though I'll definitely need help with wording for that!) and I'll change the implementation to track the region to copy by hand.

I also agree that we should handle the negative or zero `length` case like you described (add the "remainder"/the entire array if `includeRemainder` is true, return an empty array otherwise). About the `length > array.count` case though, since I documented that `length` is the *maximum* length of a chunk, should I always include the entire array in the returned result regardless of what `includeRemainder` is? Or should I document `length` as the precise length of each chunk except for the very last one (whose length can be less than or equal)?